src/main/java/org/jabref/cli/ArgumentProcessor.java

```diff
@@ -289,6 +294,66 @@ public void processArguments() {
+        }
+    }
+
+    private void checkConsistency(CliPreferences cliPreferences,
+                                  BibEntryTypesManager entryTypesManager) {
+        String fileName = cli.getCheckConsistency();
+        Optional<String> outputFormat = Optional.ofNullable(cli.getCheckConsistencyOutputFormat());
+
+        if (fileName == null) {
+            System.out.println(Localization.lang("No file specified for consistency check."));
+            return;
+        }
+
+        try {
+            Path filePath = Path.of(fileName);
+            ParserResult pr = OpenDatabase.loadDatabase(filePath, cliPreferences.getImportFormatPreferences(), fileUpdateMonitor);
+            BibDatabaseContext databaseContext = pr.getDatabaseContext();
+            List<BibEntry> entries = databaseContext.getDatabase().getEntries();
+
+            BibliographyConsistencyCheck consistencyCheck = new BibliographyConsistencyCheck();
+            BibliographyConsistencyCheck.Result result = consistencyCheck.check(entries);
+
+            if (outputFormat.isEmpty()) {
+                writeFindings(result);
+                return;
+            }
+
+            String outputFileName = fileName.substring(0, fileName.lastIndexOf('.')) + "_consistency_check." + outputFormat.get().toLowerCase();
+            try (Writer writer = new FileWriter(outputFileName, StandardCharsets.UTF_8)) {
+                if ("CSV".equalsIgnoreCase(outputFormat.get())) {
+                    BibliographyConsistencyCheckResultCsvWriter csvWriter = new BibliographyConsistencyCheckResultCsvWriter(result, writer, entryTypesManager, databaseContext.getMode());
+                    csvWriter.writeFindings();
+                } else {
+                    BibliographyConsistencyCheckResultTxtWriter txtWriter = new BibliographyConsistencyCheckResultTxtWriter(result, writer, entryTypesManager, databaseContext.getMode());
+                    txtWriter.writeFindings();
+                }
+                System.out.println(Localization.lang("Consistency check completed"));
+            }
+        } catch (IOException e) {
+            LOGGER.error("Error accessing file '{}'.", fileName);
+        }
+    }
+
+    public void writeFindings(BibliographyConsistencyCheck.Result result) {
+        System.out.print(Localization.lang("Field Presence Consistency Check Result"));
+        System.out.print("\n\n");
+
+        if (result.entryTypeToResultMap().isEmpty()) {
+            System.out.println("No errors found.\n");
+            return;
+        }
+
+        System.out.print("\n");
+        System.out.print("%s | %s\n".formatted("x", Localization.lang("required field is present")));
+        System.out.print("%s | %s\n".formatted("o", Localization.lang("optional field is present")));
+        System.out.print("%s | %s\n".formatted("?", Localization.lang("unknown field is present")));
+        System.out.print("%s | %s\n".formatted("-", Localization.lang("field is absent")));
     }
```

Comment:
Remove this method. Using `org.jabref.logic.quality.consistency.BibliographyConsistencyCheckResultTxtWriter` is enough.